src/Compilers/Core/Portable/Emit/EditAndContinue/DeltaMetadataWriter.cs

```diff
@@ -784,6 +812,266 @@ private void PopulateEncLogTableParameters()
+                        index++;
+                        continue;
+                    }
+
+                    // Either this is a new parent that hasn't had an attribute before, or we've run
+                    // out of existing rows to update for this parent. Either way, we want to add to the end of the table.
+                    rowId = nextRowId++;
+
+                    // Keep track of the addition for passing to the next generation
+                    _customAttributesAdded[rowId] = customAttributeTarget;
+                }
+
+                // If there are no more attributes for this parent to emit, then it means this was a delete of something
+                // added in the previous generation
+                if (_customAttributeParentCounts[customAttributeTarget] == 0)
+                {
+                    // Expand our enc log table to accommodate
+                    encLogRows.Add(0);
+                    logIndex = encLogRows.Count - 1;
+                    // The custom attribute we're deleting came from a previous delta, so we don't have any data about it
+                    // so we just set its constructor to a MemberRef for row id 0
+                    AddDeletedCustomAttribute(customAttributeTarget.Kind, MetadataTokens.EntityHandle(TableIndex.MemberRef, 0), value: default);
+                }
+                else
+                {
+                    _customAttributeParentCounts[customAttributeTarget]--;
+                }
+
+                if (_customAttributeParentCounts[customAttributeTarget] == 0)
+                {
+                    // Finished with this parent
+                    index++;
+                }
+
+                // remainingEmittedAttributeParents only contains records that haven't been logged, but since we know it is in the
+                // right order, we don't need to know the original index we can just fill log rows in order, and skip any
+                // that were already filled in the previous phase.
+                Debug.Assert(logIndex < encLogRows.Count, "logIndex is invalid, possibly caused by a bug in this loop specifically. Is something incrementing it twice?");
+                while (encLogRows[logIndex] != 0)
+                {
+                    logIndex++;
+                }
+
+                Debug.Assert(rowId > 0, "Shouldn't ever emit a log row for custom attribute row 0");
+                encLogRows[logIndex] = rowId;
+            }
+        }
+
+        private void AddDeletedCustomAttribute(HandleKind parentKind, EntityHandle constructor, BlobHandle value)
+        {
+            // now emit a "delete" row with a parent that is for the 0 row of the same table as the existing one
+            var isValidKind = MetadataTokens.TryGetTableIndex(parentKind, out var tableIndex);
+
+            Debug.Assert(isValidKind, "Trying to delete a custom attribute for a parent kind that doesn't have a matching table index.");
+
+            if (isValidKind)
```

Comment:
>if (isValidKind) [](http://example.com/codeflow?start=12&length=16)

I don't think this check is necessary. This might make sense if we want to gueard against the baseline having incorrect metadata but then we would need to check whether constructor has valid kind as well. I don't think we need to be that robust here though. If the baseline is corrupt and the build throws here we fail the whole EnC operation gracefully and report rude edit. Adding the check hides bugs.

Reply:
Changed this to a throw.. Still need to convert from a HandleKind to a TableIndex, and if I just ignore the result of the Try call we'd still be hiding bugs.